### ISSUE TITLE
How to compute gradient of output wrt input in Tensorflow 2.0 when there are multi-labels?

### ISSUE THREAD

**Issue by xushanthu-2014** [tensorflow/tensorflow#47172]
In my case, the input shape is (None, 16) (sample size, feature size), and my output is (None, 3). It means that my DNN is a mapping from input space to 3-D output space. In that case how can I use the tape to calculate the gradients of output w.r.t input?
Here is my code:
```
import tensorflow as tf
x_grad = tf.Variable(gpi_input, dtype=tf.float32)
with tf.GradientTape(persistent=True) as tape:
    pred = best_model(x_grad)
grads = tape.gradient(pred, x_grad)
```
where ```pred``` is of [None, 3] and ```x_grad``` is of [None, 16]. But what I got is a tensor of shape [None, 16] which means ```grads``` is of shape [None, 16]. But I want to know the gradients of each output w.r.t to each input. So what should I do?

The ```best_model``` has been already trained and we cannot retrain a model from input to first label, the second label and the third label. So how can I get gradients of every label given the model I have in my hand?

Thanks a lot!

**Comment by Saduf2019** [comment]
@xushanthu-2014 
Please provide with simple stand alone code or if possible share a colab gist with error reported.

**Comment by xushanthu-2014** [comment]
Hi @Saduf2019, the simple stand alone code see here:
```
import tensorflow as tf
sample_size, dim_feature,dim_multilabel=8000,16,3
gpi_input = np.random.normal(0, 1, size=(sample_size, dim_feature))
ground_truth = np.random.normal(0, 1, size=(sample_size, dim_multilabel))
x_grad = tf.Variable(gpi_input, dtype=tf.float32)
with tf.GradientTape(persistent=True) as tape:
    pred = best_model(x_grad)
grads = tape.gradient(pred, x_grad)
```
There is no error. But I want to get grads for 3 labels at the same time in this case of multi-label problem. How do I do that? by simply using tape.gradient I can just get grads of size [sample_size, dim_input] which means I can only get one value for 3 labels. So how can I get the derivatives of each label w.r.t each input?

Thanks

**Comment by Saduf2019** [comment]
@xushanthu-2014 
Please refer to [this link](https://github.com/dipanjanS/adversarial-learning-robustness/blob/main/tutorials/attacks/02_2_targeted_pgd.ipynb) and let us know if it helps.

**Comment by codrutartene** [comment]
@xushanthu-2014 did you managed to solve this problem? If yes, how? Thanks!

**Comment by yxteo2** [comment]
> 

I think you need to add
tape.watch(input)

And also for the prediction if you are getting the gradient of the correct output you need to select the correct output to get the gradient
